Donation-platform/src/main/java/com/google/FindTrendingCharities.java

```diff
@@ -30,144 +30,152 @@
+            this.charities = getAllCharities();
+            if (charities.size() < 1) {
+                setUp.populateDatabase();
+                this.charities = getAllCharities();
+            }
+            updateCharityScores();
+        }
     }
-    ArrayList<Charity> charitiesList = new ArrayList<>(charities);
-    Collections.sort(charitiesList);
-    List<Charity> topTrending;
-    if (charitiesList.size() > MAX_NUM_OF_CHARITIES_TO_RETURN) {
-      topTrending = charitiesList.subList(0, MAX_NUM_OF_CHARITIES_TO_RETURN);
-    } else {
-      topTrending = charitiesList;
+
+    // returns the collection of top trending charities pre-computer offline
+    public Collection<Charity> queryDb() {
+        ArrayList<Charity> charitiesList = new ArrayList<>(charities);
+        sortTrendingCharities(charitiesList);
+        return topTrendingCharities;
     }
-    return topTrending;
-  }
-
-  // returns a list of all charities in the database
-  private Collection<Charity> getAllCharities() {
-    Collection<Charity> charities = new ArrayList<>();
-    try {
-      charities = db.getAllCharities();
-    } catch (EntityNotFoundException e) {
-      System.out.println("charity entities not found: " + e);
-      return null;
-    } catch (Exception e) {
-      System.out.println("unexpected exception: " + e);
-      return null;
+
+    // returns a list of all charities in the database
+    private Collection<Charity> getAllCharities() {
+        Collection<Charity> charities = new ArrayList<>();
+        try {
+        charities = db.getAllCharities();
+        } catch (EntityNotFoundException e) {
+        System.out.println("charity entities not found: " + e);
+        return null;
+        } catch (Exception e) {
+        System.out.println("unexpected exception: " + e);
+        return null;
+        }
+        return charities;
     }
-    return charities;
-  }
-
-  // returns the trending score of inputted charity calculated
-  // as a weighted average of the tagScore and the avgReview where
-  // tagScore represents the average trending score of the associated tags
-  // and avgReview is a weighted average of the userRating and the charityNavigatory API rating
-  // Note: weights for the averages are stored as class constants
-  private double calcCharityTrendingScore(Charity charity) {
-    boolean hasCharityNavRating = hasCharityNavRating(charity);
-    double charityNavRating = 0;
-    if (hasCharityNavRating) {
-      charityNavRating = calcCharityNavRating(charity);
+
+    // returns the trending score of inputted charity calculated
+    // as a weighted average of the tagScore and the avgReview where
+    // tagScore represents the average trending score of the associated tags
+    // and avgReview is a weighted average of the userRating and the charityNavigatory API rating
+    // Note: weights for the averages are stored as class constants
+    private double calcCharityTrendingScore(Charity charity) {
```

Comment:
i thought this is now done by offline job?

Reply:
This is one of the functions only called by the offline script. See above for more details.

Reply:
then move it there

Reply:
This function relies on both the charity and tag getter methods that I won't be able to call from the script.